internal/config/instance.go

```diff
@@ -44,22 +44,18 @@ type Instance struct {
 	installSource string
 	lock          *flock.Flock
 	data          map[string]interface{}
-	// lockMutex ensures that file lock can be held only once per process.  Theoretically, this should be ensured by the `flock` package, but it isn't.  So, we need this hack.
```

Comment:
After realizing that the removal of the lock file caused problems, I removed this `lockMutex` and ensured that it still works.

Apparently, `gofrs/flock` **is** thread-safe. The actual issue that I observed when adding this comment, was due us removing the lockfile.  More details are here if interested: https://stackoverflow.com/questions/17708885/flock-removing-locked-file-without-race-condition